src/java/htsjdk/samtools/util/SimpleInterval.java

```diff
@@ -0,0 +1,45 @@
+package htsjdk.samtools.util;
+
+
+/**
+ * minimal immutable class representing a 1-based closed ended genomic interval
+ */
+public class SimpleInterval implements Locatable {
+
+    private final int start;
+    private final int end;
+    private final String contig;
+
+    public SimpleInterval(final String contig, final int start, final int end){
+        if(contig == null){
+            throw new IllegalArgumentException("contig cannot be null");
```

Comment:
Now we are getting philosophical... to be clear, I am fine with Louis's solution of having `SAMRecord.getInterval()` return null for unmapped reads rather than providing a way for `SimpleInterval` itself to represent the "location unknown" state. I just wanted to bring up the old GATK approach of having `isUnmapped()` built into the `GenomeLoc` class to make sure that we don't want that here.